homeassistant/components/blink/config_flow.py

```diff
@@ -0,0 +1,109 @@
+"""Config flow to configure Blink."""
+import logging
+
+from blinkpy import blinkpy
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_PASSWORD,
+    CONF_PIN,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+
+from .const import BLINK_CONFIG, DEFAULT_OFFSET, DEFAULT_SCAN_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class BlinkConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Blink config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize the blink flow."""
+        self.blink = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+
+        stored_username = ""
+        stored_password = ""
+        stored_interval = DEFAULT_SCAN_INTERVAL
+
+        if BLINK_CONFIG in self.hass.data:
+            stored_username = self.hass.data[BLINK_CONFIG].get(CONF_USERNAME)
+            stored_password = self.hass.data[BLINK_CONFIG].get(CONF_PASSWORD)
+            stored_interval = self.hass.data[BLINK_CONFIG].get(CONF_SCAN_INTERVAL)
```

Comment:
We shouldn't do this. Imports need to get their data passed in.

Reply:
If this data is coming from YAML, it should be passed in as adata when you initialize the import. Otherwise there should be no data stored in `hass.data` that is used during a config flow. Config flow instances are unique per flow so you can store things as instance variables.